openpype/client/server/entities.py

```diff
@@ -199,22 +212,51 @@ def get_assets(
         active = None
 
     con = get_server_api_connection()
+
     fields = folder_fields_v3_to_v4(fields, con)
     kwargs = dict(
         folder_ids=asset_ids,
-        folder_names=asset_names,
         parent_ids=parent_ids,
         active=active,
-        fields=fields
     )
+    if not asset_names:
+        for folder in _folders_query(project_name, con, fields, **kwargs):
+            yield convert_v4_folder_to_v3(folder, project_name)
+        return
 
-    if fields is None or "tasks" in fields:
-        folders = get_folders_with_tasks(con, project_name, **kwargs)
+    new_asset_names = set()
+    folder_paths = set()
+    if asset_names:
```

Comment:
It would be assumed that this would never reach if line `222` was true (so we do have `asset_names`), right?

Reply:
Yep, you're right.